src/Storage.ts

```diff
@@ -15,16 +15,16 @@ export default class Storage<T> {
     this.key = key;
   }
 
-  read(): Promise<PersistedData<T>> {
-    return this.storage.getItem(this.key);
+  async read(): Promise<PersistedData<T>> {
+    return await this.storage.getItem(this.key);
```

Comment:
I think it's enough to just `return` here since `async` functions wrap a `Promise.resolve()` over the returned value.

Reply:
Ah, very good point! Will fix.